Dynatrace how to (#1094)
Kong/developer.konghq.com: app/_how-tos/set-up-dynatrace-with-otel.md

    - plugins
    - partnership

prereqs:
  entities:
    services:
        - example-service
    routes:
        - example-route
  gateway:
    - name: KONG_TRACING_INSTRUMENTATIONS
    - name: KONG_TRACING_SAMPLING_RATE
  inline:
  - title: Environment variables for Dynatrace
    position: before
    content: |

>       Set the following {{site.base_gateway}} configuration that's required for Dynatrace:

Author: @Guaris Since this wording only shows for Gateway here, they configure the data plane later with the quickstart and pass in these env var, so I'll reword this so they know they export these before doing the data plane

      {: data-deployment-topology="on-prem" }
      ```sh
      export KONG_TRACING_INSTRUMENTATIONS=all
      export KONG_TRACING_SAMPLING_RATE=1.0
      ```
      {: data-deployment-topology="on-prem" }

>       When you create the Data Plane in {{site.konnect_short_name}}, append the following {{site.base_gateway}} configuration that's required for Dynatrace:

Reviewer: ```suggestion
      When you create the Data Plane in {{site.konnect_short_name}}, add the following {{site.base_gateway}} configuration variables for Dynatrace:
```

      {: data-deployment-topology="konnect" }
      ```sh
      -e "KONG_TRACING_INSTRUMENTATIONS=all" \
      -e "KONG_TRACING_SAMPLING_RATE=1.0" \
      ```
      {: data-deployment-topology="konnect" }
  - title: Dynatrace
    content: |
      This tutorial requires you to have a Dynatrace SaaS account.

      1. In Dynatrace, find your [environment ID](https://docs.dynatrace.com/docs/discover-dynatrace/get-started/monitoring-environment#environment-id).
      1. [Generate an API token](https://docs.dynatrace.com/docs/discover-dynatrace/references/dynatrace-api/basics/dynatrace-api-authentication#create-token) with the `openTelemetryTrace.ingest` and `metrics.ingest` scopes.

      Export those values as environment variables:
      ```sh

>       export DECK_DYNATRACE_ENVRIONMENT_ID=<environment-id-here>

Reviewer: ```suggestion
      export DECK_DYNATRACE_ENVIRONMENT_ID=<environment-id-here>
```

    a: This error is because you need to add the [Dynatrace API token as an Authorization header](https://docs.dynatrace.com/docs/ingest-from/opentelemetry/getting-started/otlp-export#authentication-export-to-activegate) when you configure the OpenTelemetry plugin. 

---

## 1. Enable the OTEL plugin

In this tutorial, we'll be configuring the OpenTelemetry plugin to send {{site.base_gateway}} traces and logs to Dynatrace SaaS. This configuration is good for testing purposes, but we recommend using a collector, like [Dynatrace Collector](https://docs.dynatrace.com/docs/ingest-from/opentelemetry/collector), in production environments.

Enable the OTEL plugin with Dynatrace settings configured:

{% entity_examples %}
entities:
  plugins:
  - name: opentelemetry
    config:
      traces_endpoint: "https://${dynatrace_environment_id}.live.dynatrace.com/api/v2/otlp/v1/traces"
      logs_endpoint: "https://${dynatrace_environment_id}.live.dynatrace.com/api/v2/otlp/v1/logs"
      resource_attributes:
        service.name: "kong-dev"
      headers:
        Authorization: Api-Token ${dynatrace_api_token}

variables:
  dynatrace_environment_id:

>     value: $DYNATRACE_ENVRIONMENT_ID

Reviewer: ```suggestion
    value: $DYNATRACE_ENVIRONMENT_ID
```
